### PR TITLE
Fix security context

### LINKED ISSUE #69: [Openshift] Secrets webhook breaks default SCC on 1.15.2
**Describe the bug**:

On Openshift, after upgrading from 1.15.1 to 1.15.2, the injector stops working correctly because initContainers are rejected by default security context constraint  ("restricted").

**Steps to reproduce the bug**:
Run the webhook 1.15.2 on Openshift 4.X

**Why**

This is introduced by : 
https://github.com/banzaicloud/bank-vaults/pull/1562
https://github.com/banzaicloud/bank-vaults/commit/bd4da3a0397033c2a908ac02897a230812a489d2#diff-32802053da07acef355a6beb95cc5c4494d6d9c77ee709fb9093c7febb4bfc05L51

The init container is now rejected because : 
* It runs with user "0" even with `vault.security.banzaicloud.io/run-as-non-root: "true"`  - which is not tolerated by Openshift default "restricted" SCC
* You cannot explicitely add any capability with Openshift default "restricted" SCC

Not sure what the solution is : 
* Would it be possible to add an annotation that removes completely the security context from the specs?  (would make sense in an Openshift environment, providing secure defaults)
* If so, would the Openshift default capabilities be compatible with the secret webhook requirements? (reminder: it works fine with 1.15.1)
* If not...  this is kinda of problem, because that would require to use a custom SCC for _each project consuming secrets_. Not really an option.

Thanks,

/kind bug

### PR DESCRIPTION
**Pull request by sagikazarmark**
## Overview

Fixes #69

### CODE REVIEW

File: e2e/main_test.go [hunk 1]
```diff
@@ -90,23 +94,28 @@ func TestMain(m *testing.M) {
...
-	// testenv.Setup(envfuncs.CreateNamespace(ns))
-	// testenv.Finish(envfuncs.DeleteNamespace(ns))
+		// Set up test namespace
+		// ns := envconf.RandomName("webhook-test", 16)
+		// testenv.Setup(envfuncs.CreateNamespace(ns))
+		// testenv.Finish(envfuncs.DeleteNamespace(ns))
 
-	// Unsealing and Vault access only works in the default namespace at the moment
-	testenv.Setup(useNamespace("default"))
+		// Unsealing and Vault access only works in the default namespace at the moment
```
Comment (ramizpolic) [inline e2e/main_test.go:110]: Should we consider adding the feature to support different namespaces (maybe even on cluster-wide level)?

**Review by ramizpolic — APPROVED** [review]

File: e2e/deploy/vault-secrets-webhook/values.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
+env:
+  VAULT_IMAGE: hashicorp/vault:1.14.1
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply e2e/deploy/vault-secrets-webhook/values.yaml:2]: We may want to in the future, but I think it's enough for now. It's only used for the vault agent test.

File: e2e/main_test.go [hunk 3]
```diff
@@ -58,8 +59,11 @@ func TestMain(m *testing.M) {
 	}
 	log.SetLogger(klog.NewKlogr())
 
+	bootstrap := strings.ToLower(os.Getenv("BOOTSTRAP")) != "false"
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply e2e/main_test.go:62]: Yes, you probably want bootstrapping even with a real cluster when running these tests.

File: e2e/main_test.go (hunk 1 above)
Comment (sagikazarmark) [reply e2e/main_test.go:110]: It's actually just misconfiguration in the policies. Yes, tests should be able to run in random namespaces, but it's out of scope for this PR.

Comment (sagikazarmark) [reply e2e/main_test.go:110]: #161

**Pull request merged**